crates/ursa-network/src/origin.rs

```diff
@@ -0,0 +1,178 @@
+use std::collections::VecDeque;
+use std::fmt::{Debug, Display};
+use std::slice::Iter;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use anyhow::{anyhow, Result};
+use cid::Cid;
+use db::Store;
+use fnv::FnvHashMap;
+use futures_util::FutureExt;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_car::load_car;
+use libp2p::{
+    core::connection::ConnectionId,
+    swarm::{
+        behaviour::FromSwarm, dummy::ConnectionHandler, NetworkBehaviour, NetworkBehaviourAction,
+        PollParameters,
+    },
+    PeerId,
+};
+use surf::Client;
+use tokio::sync::oneshot::{channel, Receiver};
+use tracing::error;
+use void::Void;
+
+use ursa_store::UrsaStore;
+
+use crate::OriginConfig;
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub enum Origin {
+    Ipfs,
+    Arweave,
+    Filecoin,
+}
+
+impl Origin {
+    pub fn iter() -> Iter<'static, Origin> {
+        [Origin::Ipfs].iter()
+    }
```

Comment:
Is this used?

Reply:
Good catch! I initially had an implementation where providing origin was optional, and if unprovided we could easily iterate over the list of supported origins and try them one by one using this, but instead went with a specific origin per request. Will remove this :+1:

Reply:
https://github.com/fleek-network/ursa/pull/278/commits/c329178e1f5d47d673bdcf7a65187e8fcd02bd66